src/main/java/org/spongepowered/common/registry/type/text/TextSerializerRegistryModule.java

```diff
@@ -65,20 +47,11 @@ public void registerDefaults() {
     }
 
     private void registerSerializer(TextSerializer serializer) {
-        this.textSerializerMappings.put(serializer.getId(), serializer);
+        this.map.put(serializer.getId(), serializer);
     }
 
     @Override
     public void registerAdditionalCatalog(TextSerializer serializer) {
-        this.textSerializerMappings.put(serializer.getId(), serializer);
-    }
-
-    @Override
-    public Map<String, TextSerializer> provideCatalogMap() {
-        final HashMap<String, TextSerializer> map = new HashMap<>();
-        for (Map.Entry<String, TextSerializer> entry : this.textSerializerMappings.entrySet()) {
-            map.put(entry.getKey().replace("minecraft:", "").replace("sponge:", ""), entry.getValue());
-        }
-        return map;
+        this.map.put(serializer.getId(), serializer);
```

Comment:
This should check whether the id is registered already and lowercase the id to prevent it from becoming inaccessible.

Reply:
Also missing null check